app/src/main/java/com/wesleyfranks/todo24/data/TodoRepository.kt

```diff
@@ -22,23 +23,23 @@ class TodoRepository {
         initializeDB(context)
 
         CoroutineScope(IO).launch {
-            todoDatabase!!.TodoDao()!!.insert(todo)
+            todoDatabase!!.TodoDao().insert(todo)
```

Comment:
You should just get a reference to the Dao once, and then you don't have to assert that todoDatabase isn't null, and then grab the Dao from it every time. 
Something at the top:
```
lateinit var todoDatabase: TodoDatabase
lateinit var dao: TodoDao
```
In initializeDB:
```
dao = todoDatabase.TodoDao()
```
Everywhere else:
```
dao.insert(todo)
```